src/web/components/App.tsx

```diff
@@ -329,6 +331,32 @@ export const App = ({ CAPI, NAV, ophanRecord }: Props) => {
 		}
 	}, [countryCode, CAPI.config.switches.consentManagement, pageViewId]);
 
+	// ************************
+	// *      Commercial      *
+	// ************************
+	useEffect(() => {
+		if (!window.guardian.config.switches.commercialMetrics) return;
+		if (!window.guardian.config.ophan) return;
+
+		const shouldForceMetrics = ABTestAPI.allRunnableTests(
+			tests,
+		).some((test) => test.id.startsWith('Commercial'));
+
+		const userIsInSamplingGroup = Math.random() <= 1 / 100;
+
+		log('commercial', { shouldForceMetrics, userIsInSamplingGroup });
+
+		const { isDev } = window.guardian.config.page;
+
+		if (isDev || shouldForceMetrics || userIsInSamplingGroup) {
+			// we’re doing this twice
+			const browserId = getCookie('bwid') || undefined;
+
+			// TODO: only send them once, using the return value of sendBeacon
```

Comment:
Thanks Max!
Since we're sending metrics twice, would it make sense to add a static field to `EventTimer` (something like `commercialMetricsSent` which could be either `true` or `false`, and set the value to `true` when metrics are first sent? Since `EventTimer` is a singleton and we always retrieve the same instance with `EventTimer.get()` this should solve the problem of sending duplicate metrics.

Reply:
Yeah, it’s a draft, we can used the same `logged` value, but I need to first figure out how state works in React!

Reply:
Should be fixed now!